src/zkp/pifac.rs

```diff
@@ -24,43 +33,62 @@ use std::fmt::Debug;
 use tracing::error;
 use zeroize::ZeroizeOnDrop;
 
+/// Proof that the modulus N can be factored into two numbers greater than 2^ell
+/// for a parameter ell.
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub(crate) struct PiFacProof {
-    P: Commitment,
-    Q: Commitment,
-    A: Commitment,
-    B: Commitment,
-    T: Commitment,
+    /// Commitment to the factor `p` (`P` in the paper).
+    p_commitment: Commitment,
+    /// Commitment to the factor `q` (`Q` in the paper).
+    q_commitment: Commitment,
+    /// Commitment to randomness alpha and `x` (`A` in the paper).
+    p_mask_commitment: Commitment,
+    /// Commitment to randomness beta and `y` (`B` in the paper).
+    q_mask_commitment: Commitment,
+    /// Combination of commitment to `Q` using ring Pedersen parameter `r` (`T`
+    /// in the paper).
+    // Commitment to q + p's commitment randomness
+    q_link_commitment: Commitment,
+    /// Randomness for commitment.
     sigma: CommitmentRandomness,
```

Comment:
suggestion: maybe call this `link_randomness` or something to make it clearer which commitment it's supposed to correspond to.